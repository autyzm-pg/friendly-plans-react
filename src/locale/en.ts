const translations = {
  planActivity: {
    planNamePlaceholder: 'Type plan name',
    shuffleTasks: 'Shuffle tasks',
  },
  common: {
    cancel: 'Cancel',
    ok: 'OK',
    yes: 'Yes',
    email: 'Email address',
    name: 'Your Name',
    password: 'Password',
    error: 'Error',
    success: 'Success',
    unknownError: 'Unknown error occured',
    addImage: 'Add Image',
  },
  validation: {
    email: 'Must be a valid email',
    required: 'This field is required',
    passwordLength: 'Password is too short',
    acceptRequired: 'You need to accept our Terms of Use',
    duplicatedPlan: 'Plan name already exists',
    planNameRequired: 'Plan name is required',
  },
  notifications: {
    channelName: 'Notifications',
    channelDescription: 'Get urgent notifications about the service',
  },
  signIn: {
    signIn: 'Sign In',
    signInButton: 'Sign In',
    signInAsAnonymous: 'Use the App as Guest',
    signUpTip: "don't have an account?",
    anonymousTip: 'you can also',
    forgotPassword: 'forgot password?',
  },
  signUp: {
    signUp: 'Sign Up',
    signUpButton: 'Sign up',
    accept: 'I accept ',
    termsOfUse: 'Terms of Use',
  },
  resetPassword: {
    resetPassword: 'Reset password',
    guide1: 'Provide you email in the field below. We will send you link to reset your password.',
    guide2: 'After setting up new password, sign in with newly set password at the login screen.',
    resetPasswordButtonLabel: 'Reset Password',
    userNotFound: 'No user found for provided email. Make sure to provide email you registered.',
    resetPasswordSuccess:
      'Email with password reset link has been sent. After setting up new password, sign in with newly set password at the login screen.',
  },
  content: {
    content: 'Content',
    contentList: 'Content List',
  },
  settings: {
    settings: 'Settings',
    signOutTitle: 'Sign out',
    signOutSubtitle: 'After you sign out, you will need to sign in again.',
    signOutAction: 'Sign out',
    signOutDialogTitle: 'Sign out',
    signOutDialogDescription: 'Are you sure you want to sign out?',
  },
  sidebar: {
    addTask: 'Add SimpleTask',
    addStudent: 'Add Student',
    takeAPicture: 'Take a Picture',
    recordSound: 'Record Sound',
  },
  studentList: {
    createStudent: 'Add a Student',
    screenTitle: 'Select student',
    search: 'Search',
    dashboard: 'Student dashboard',
    studentNamePlaceholder: 'Enter Student name...',
    removeStudentTitle: 'Confirm',
    removeStudentDescription:
      'Are you sure you want to remove {{name}} from your student list?\n This action cannot be undone.',
  },
  planList: {
    viewTitle: 'All Plans',
    createPlan: 'Add new plan',
    copyPlan: 'Copy existing plan',
    conjunction: 'lub',
    planNamePlaceholder: 'Muzykowanie',
  },
  planItemActivity: {
    viewTitleTask: 'Task',
    viewTitleInteraction: 'Interaction',
    viewTitleBreak: 'Break',
    taskNamePlaceholder: 'Type task name',
    timerButton: 'Set up timer',
    taskNameForChild: 'Type task name for child',
    addImage: 'Add image',
    imageActionTakePhoto: 'Take a photo',
    imageActionLibrary: 'Browse from library',
    imageActionBrowse: 'Browse from device',
    imageLibraryTitle: 'Image Library',
  },
  updatePlan: {
    screenTitle: '{{studentName}} - Update a plan',
    removePlanTitle: 'Confirm',
    removePlanDescription:
      'Are you sure you want to plan "{{name}}" from student plan list?\n This action cannot be undone.',
    addBreak: 'Add break',
    addInteraction: 'Add interaction',
    addTask: 'Add task',
    planItemNamePlaceholder: 'Enter Plan Item name...',
  },
  studentSettings: {
    studentName: "Student's name",
    taskView: "Task's view",
    soundSettings: 'Sound settings',
    settingsTitle: "Student's settings",
    createStudentTitle: 'Create a Student',
    textSettingsSizeS: 'Font size: S',
    textSettingsSizeM: 'Font size: M',
    textSettingsSizeL: 'Font size: L',
    textSettingsSizeXL: 'Font size: XL',
    largeImageSlide: 'Large image as a slide',
    imageWithTextSlide: 'Image with text label as a slide',
    textSlide: 'Just text as a slide',
    imageWithTextList: 'image with text label as a list',
    textList: 'Just text label as a list',
    uppercase: 'Uppercase letters',
    blockSwipe: 'Block swipe',
    alarmSound: 'Alarm sound',
    planCardPlacehorder: "Let's play paper, rock, scissors",
    studentNamePlaceholder: 'Enter student name',
    removeStudent: 'Remove student',
    createStudent: 'Create student',
  },
  runPlan: {
    next: 'Next',
    wait: 'Wait...',
  },
  taskTable: {
    number: '#',
    name: 'Name',
    type: 'Type',
    section: 'Parts',
    time: 'Time',
    delete: 'Delete',
    edit: 'Edit',
  },
};

export default translations;
