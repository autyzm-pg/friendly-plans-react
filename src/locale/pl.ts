const translations = {
  planActivity: {
    planNamePlaceholder: 'Wpisz nazwę planu',
    shuffleTasks: 'Tasuj zadania',
  },
  common: {
    cancel: 'Anuluj',
  },
  signIn: {
    signIn: 'Zaloguj się',
    signInAsAnonymous: 'Zaloguj się anonimowo',
  },
  content: {
    content: 'Treść',
    contentList: 'Treści',
  },
  settings: {
    settings: 'Ustawienia',
    signOutTitle: 'Wyloguj się',
    signOutSubtitle: 'Po wylogowaniu się, musisz zalogować się ponownie',
    signOutAction: 'Wyloguj',
    signOutDialogTitle: 'Wyloguj się',
    signOutDialogDescription: 'Czy na pewno chcesz się wylogować?',
  },
  planList: {
    viewTitle: 'Wszystkie plany',
    createPlan: 'DODAJ NOWY PLAN',
    copyPlan: 'SKOPIUJ ISTNIEJĄCY PLAN',
    conjunction: 'lub',
    planNamePlaceholder: 'Muzykowanie',
  },
  planItemActivity: {
    viewTitleTask: 'Zadanie',
    viewTitleInteraction: 'Interakcja',
    viewTitleBreak: 'Przerwa',
    taskNamePlaceholder: 'Wpisz nazwę zadania',
    timerButton: 'Ustaw timer',
    taskNameForChild: 'Nazwij dla dziecka',
    addImage: 'Dodaj obrazek',
    imageActionTakePhoto: 'Zrób zdjęcie',
    imageActionLibrary: 'Wybierz z biblioteki',
    imageActionBrowse: 'Dodaj z urządzenia',
    imageLibraryTitle: 'Biblioteka zdjęć',
  },
  updatePlan: {
    addBreak: 'Dodaj przerwę',
    addInteraction: 'Dodaj interakcję',
    addTask: 'Dodaj zadanie',
  },
  studentSettings: {
    studentName: 'Imię podopiecznego',
    taskView: 'Widok zadań',
    soundSettings: 'Ustawienia dźwięku',
    settingsTitle: 'Ustawienia podopiecznego',
    createStudentTitle: 'Stwórz Studenta',
    textSettingsSizeS: 'Rozmiar liter: S',
    textSettingsSizeM: 'Rozmiar liter: M',
    textSettingsSizeL: 'Rozmiar liter: L',
    textSettingsSizeXL: 'Rozmiar liter: XL',
    largeImageSlide: 'Duży obrazek jako slajd',
    imageWithTextSlide: 'Obrazek z tekstem jako slajd',
    textSlide: 'Tylko tekst jako slajd',
    imageWithTextList: 'Obrazek z tekstem jako lista',
    textList: 'Tylko tekst jako lista',
    uppercase: 'Wielkie litery',
    blockSwipe: "Blokada swipe'u",
    alarmSound: 'Dźwięk alarmu',
    planCardPlacehorder: 'Zagraj w papier, kamień, nożyce',
    studentNamePlaceholder: 'Wpisz imię podopiecznego',
    removeStudent: 'Usuń studenta',
    createStudent: 'Dodaj Studenta',
  },
  studentsList: {
    screenTitle: 'Wybór podopiecznego',
    search: 'Wyszukaj',
    dashboard: 'Studenci',
    createStudent: 'Stwórz studenta',
  },
  runPlan: {
    next: 'Dalej',
    wait: 'Czekaj...',
  },
  taskTable: {
    number: '#',
    name: 'Nazwa',
    type: 'Typ',
    section: 'Części',
    time: 'Czas',
    delete: 'Usuń',
    edit: 'Edycja',
  },
  validation: {
    duplicatedPlan: 'Plan o podanej nazwie już istnieje',
    planNameRequired: 'Nazwa planu jest wymagana',
  },
};

export default translations;
