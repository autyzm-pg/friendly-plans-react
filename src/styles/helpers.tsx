import { ViewStyle } from 'react-native';
import { palette } from './palette';

export const getElevation = (elevation: number): ViewStyle => ({
  elevation,
  shadowColor: palette.shadow,
  shadowOffset: {
    width: 0,
    height: Math.round(1 + elevation / 4),
  },
  shadowOpacity: 0.17 + elevation / 100,
  shadowRadius: 1 + elevation / 2,
});
