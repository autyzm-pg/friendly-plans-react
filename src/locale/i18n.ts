import i18n, { LanguageDetectorModule } from 'i18next';
import { initReactI18next } from 'react-i18next';
import { getLocales } from 'react-native-localize';

import en from './en';
import pl from './pl';

const languageDetector: LanguageDetectorModule = {
  type: 'languageDetector',
  init(services, detectorOptions, i18nextOptions) {
    /* use services and options */
  },
  detect() {
    return getLocales()[0].languageCode;
  },
  cacheUserLanguage(lng) {
    /* cache language */
  },
};

i18n
  .use(languageDetector)
  .use(initReactI18next)
  .init({
    fallbackLng: 'en',
    resources: {
      en,
      pl,
    },
    ns: ['common', 'signIn', 'signUp'],
    defaultNS: 'common',
    debug: __DEV__,
    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
