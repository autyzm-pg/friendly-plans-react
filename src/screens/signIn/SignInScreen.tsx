import React from 'react';
import { StyleSheet } from 'react-native';
import { NavigationInjectedProps } from 'react-navigation';

import { FlatButton, FullScreenTemplate, StyledText } from 'components';
import { i18n } from 'locale';
import { palette, typography } from 'styles';
import { SignInAsAnonymousButton } from './SignInAsAnonymousButton';
import { SignInFormContainer } from './SignInFormContainer';

export class SignInScreen extends React.PureComponent<NavigationInjectedProps> {
  navigateToSignUp = () => this.props.navigation.navigate('SignUp');

  render() {
    return (
      <FullScreenTemplate padded narrow>
        <SignInFormContainer />

        <StyledText style={styles.signUpTip}>{i18n.t('signIn:signUpTip')}</StyledText>
        <FlatButton onPress={this.navigateToSignUp} title={i18n.t('signUp:signUpButton')} />

        <StyledText style={styles.anonymousTip}>{i18n.t('signIn:anonymousTip')}</StyledText>
        <SignInAsAnonymousButton />
      </FullScreenTemplate>
    );
  }
}

const styles = StyleSheet.create({
  signUpTip: {
    ...typography.caption,
    color: palette.textDisabled,
    textAlign: 'center',
    marginTop: 4,
  },
  anonymousTip: {
    ...typography.caption,
    color: palette.textDisabled,
    textAlign: 'center',
    marginTop: 20,
  },
});
