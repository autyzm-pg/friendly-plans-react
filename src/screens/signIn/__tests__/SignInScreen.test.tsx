import { shallow } from 'enzyme';
import React from 'react';

import { navigationMock } from 'helpers';
import { i18n } from 'locale';
import { SignInScreen } from '../SignInScreen';

describe('SignInScreen', () => {
  const props = {
    navigation: navigationMock,
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<SignInScreen {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should navigate to signup on sign up button press', () => {
    const wrapper = shallow(<SignInScreen {...props} />);
    const button = wrapper.find(`Button[title="${i18n.t('signUp:signUp')}"]`);
    button.simulate('press');
    expect(props.navigation.navigate).toBeCalledWith('SignUp');
  });
});
