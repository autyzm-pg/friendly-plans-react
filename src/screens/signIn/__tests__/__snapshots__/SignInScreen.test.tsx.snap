// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInScreen should match snapshot 1`] = `
<FullScreenTemplate
  narrow={true}
  padded={true}
>
  <SignInFormContainer />
  <StyledText
    style={
      Object {
        "color": "#9B9B9B",
        "fontFamily": "Rubik-Regular",
        "fontSize": 12,
        "letterSpacing": 0.4,
        "marginTop": 4,
        "textAlign": "center",
      }
    }
  >
    don't have an account?
  </StyledText>
  <FlatButton
    onPress={[Function]}
    title="Sign up"
  />
  <StyledText
    style={
      Object {
        "color": "#9B9B9B",
        "fontFamily": "Rubik-Regular",
        "fontSize": 12,
        "letterSpacing": 0.4,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    you can also
  </StyledText>
  <SignInAsAnonymousButton />
</FullScreenTemplate>
`;
